Editor/HPUIInteractorConeRayAnglesEditor.cs

```diff
@@ -0,0 +1,155 @@
+
+        private const string DONT_ASK_FOR_OVERRIDE_EDITOR_PREF_KEY = "ubco.ovilab.HPUI.Interaction.Components.ConeThresholdsEditor.DontAskForOverride";
+        private bool userSelection;
+        private string saveName;
+
+        private bool advancedSettingsFoldout;
+
+        protected void OnEnable()
+        {
+            t = target as HPUIInteractorConeRayAngles;
+            if (!t) return;
+            foreach ((XRHandJointID phalange, List<HPUIInteractorRayAngle> rays) in t.ActiveFingerAngles)
+            {
+                if (!baselineThresholds.ContainsKey(phalange) && rays.Count > 0)
+                {
+                    baselineThresholds[phalange] = rays.Select(ray => ray.RaySelectionThreshold).ToList();
+                }
+            }
+        }
+
+        public override void OnInspectorGUI()
+        {
+            base.OnInspectorGUI();
+
+            EditorGUILayout.LabelField("", GUI.skin.horizontalSlider);
+
+            advancedSettingsFoldout = EditorGUILayout.Foldout(advancedSettingsFoldout, "Advanced Settings", toggleOnLabelClick:true);
+            if (advancedSettingsFoldout)
+            {
+                EditorGUILayout.LabelField("Apply Ray Selection Threshold Offset", EditorStyles.boldLabel);
+                offsetValue = EditorGUILayout.FloatField("Offset", offsetValue);
+                applyToAll = EditorGUILayout.Toggle("Apply to All", applyToAll);
+                overwriteThisAsset = EditorGUILayout.Toggle("Overwrite This Asset", overwriteThisAsset);
+
+                EditorGUI.BeginDisabledGroup(applyToAll);
+                selectedPhalange = (XRHandJointID)EditorGUILayout.EnumPopup("Phalange", selectedPhalange);
+                EditorGUI.EndDisabledGroup();
+
+                if (overwriteThisAsset) EditorGUILayout.HelpBox("You are about to overwrite this asset!", MessageType.Warning); // warning
+                if (overwriteThisAsset) GUI.backgroundColor = Color.red; // more warning
+                if (GUILayout.Button(overwriteThisAsset
+                        ? "Overwrite this Asset and Apply Offset"
+                        : "Create Copy and Apply Offset"))
+                {
+                    if (overwriteThisAsset)
+                    {
+                        userSelection = EditorUtility.DisplayDialog("Overwrite Cone Ray Angles Asset",
+                            "Are you sure you want to overwrite this asset?" +
+                            "\nYou can undo or set offsets to 0 to regain original values." +
+                            "\nOnce you overwrite and close the editor the values will be permanently modified!",
+                            "Yes", "No", DialogOptOutDecisionType.ForThisSession,
+                            DONT_ASK_FOR_OVERRIDE_EDITOR_PREF_KEY); // https://www.youtube.com/watch?v=xrg-RgF5F8o Warning
+                        if (userSelection) ApplyModifications();
```

Comment:
Make this in a different line in a {} block, it makes it clear whats happening here.